app/controllers/posts_controller.rb

```diff
@@ -11,5 +12,23 @@ def show
     @user = User.find(user_id)
     @post = Post.find(post_id)
     @comments = Comment.where(post_id:)
+    @recent_user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    user = current_user
+    puts user
+    @post = Post.new(
+      title: params[:post][:title],
+      text: params[:post][:text],
+      author: user
+    )
```

Comment:
[OPTIONAL] I think your app will be safer if it uses strong parameters. You can check this [link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) out